trade-finance-manager-ui/server/controllers/case/activity/index.js

```diff
@@ -76,12 +86,21 @@ const filterActivities = async (req, res) => {
 
   const activities = mapActivities(deal.tfm.activities);
 
+  const { dealSnapshot } = deal;
+
+  const successMessage = await getDealSuccessBannerMessage({
```

Comment:
I tried this & unfortunately it seems to need the context of the `req` object- I asume req.flash calls `this.session` or something similar :(

Reply:
Did you try updating `getFlashSuccessMessage` to this?

```ts
export const getFlashSuccessMessage = (reqFlash) => 

// getFlashSuccessMessage(req.flash);
```

Really want to avoid passing the `req` around functions as it is not a good practice

Reply:
Yep in commit 501917d47574d95c7ce89f7bfe8f95c2a6571809 I tried this, but it was throwing `'req.flash() requires sessions'` - looking [at the connect flash package](https://github.com/jaredhanson/connect-flash/blob/master/lib/flash.js) it uses `this.session`. I could probably get around this by passing in `flash.bind(req)` but I think thats replacing one code smell with another

Reply:
Tbh the more I look at connect-flash the more I dislike it - it was last updated 11 years ago :'( https://www.npmjs.com/package/connect-flash

Reply:
Completely agree, not ideal :/ Thinking of cleaner ways to do this, what do you think about this?

```ts
  const successMessage = await getDealSuccessBannerMessage({
    dealSnapshot,
    userToken,
    // req,
    flashSuccessMessage: getFlashSuccessMessage(req),
  });
```

Or maybe even this - it just repeats `getFlashSuccessMessage` - it is not DRY, but the positive is we're not passing `req` to another function and is most likely more efficient. If we were to do this I think removing `getFlashSuccessMessage` to be consistent would be better.

```ts
  const successMessage = await getDealSuccessBannerMessage({
    dealSnapshot,
    userToken,
    // req,
    flashSuccessMessage: req.flash('successMessage')[0],
  });
```

Just really want to avoid passing `req` around, it passes _so_ many things into a function when the function typically only needs 1x thing

Reply:
Have updated, I do feel that `getDealSuccessBannerMessage` is doing something slightly odd now by returning a parameter passed into it but happy to leave & get this merged in.
Also added some more constants and types so that ` flashSuccessMessage: req.flash(FLASH_TYPES.SUCCESS_MESSAGE)[0],` feels a bit safer. Tried to get declaration merging working to override the flash type & only accept `FlashType` rather than string, but wasn't quite working & non-essential

Reply:
Great work @MarRobSoftwire 

I do agree it's a bit odd, but it seems to be the cleanest approach we can do without passing `req` around in different functions